pkg/applicationmanager/controllers/applicationbackupschedule.go

```diff
@@ -0,0 +1,342 @@
+			s.Recorder.Event(backupSchedule,
+				v1.EventTypeWarning,
+				string(stork_api.ApplicationBackupStatusFailed),
+				msg)
+			log.ApplicationBackupScheduleLog(backupSchedule).Error(msg)
+			return err
+		}
+
+		if backupSchedule.Spec.Suspend == nil || !*backupSchedule.Spec.Suspend {
+			// Then check if any of the policies require a trigger
+			policyType, start, err := s.shouldStartApplicationBackup(backupSchedule)
+			if err != nil {
+				msg := fmt.Sprintf("Error checking if backup should be triggered: %v", err)
+				s.Recorder.Event(backupSchedule,
+					v1.EventTypeWarning,
+					string(stork_api.ApplicationBackupStatusFailed),
+					msg)
+				log.ApplicationBackupScheduleLog(backupSchedule).Error(msg)
+				return err
+			}
+
+			// Start a backup for a policy if required
+			if start {
+				err := s.startApplicationBackup(backupSchedule, policyType)
+				if err != nil {
+					msg := fmt.Sprintf("Error triggering backup for schedule(%v): %v", policyType, err)
+					s.Recorder.Event(backupSchedule,
+						v1.EventTypeWarning,
+						string(stork_api.ApplicationBackupStatusFailed),
+						msg)
+					log.ApplicationBackupScheduleLog(backupSchedule).Error(msg)
+					return err
+				}
+			}
+		}
+
+		// Finally, prune any old backups that were triggered for this
+		// schedule
+		err = s.pruneApplicationBackups(backupSchedule)
+		if err != nil {
+			msg := fmt.Sprintf("Error pruning old backups: %v", err)
+			s.Recorder.Event(backupSchedule,
+				v1.EventTypeWarning,
+				string(stork_api.ApplicationBackupStatusFailed),
+				msg)
+			log.ApplicationBackupScheduleLog(backupSchedule).Error(msg)
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *ApplicationBackupScheduleController) setDefaults(backupSchedule *stork_api.ApplicationBackupSchedule) {
+	if backupSchedule.Spec.ReclaimPolicy == "" {
+		backupSchedule.Spec.ReclaimPolicy = stork_api.ReclaimPolicyRetain
+	}
+}
+
+func getApplicationBackupStatus(name string, namespace string) (stork_api.ApplicationBackupStatusType, error) {
+	backup, err := k8s.Instance().GetApplicationBackup(name, namespace)
+	if err != nil {
+		return stork_api.ApplicationBackupStatusFailed, err
+	}
+
+	return backup.Status.Status, nil
+}
+
+func (s *ApplicationBackupScheduleController) updateApplicationBackupStatus(backupSchedule *stork_api.ApplicationBackupSchedule) error {
+	updated := false
+	for _, policyApplicationBackup := range backupSchedule.Status.Items {
+		for _, backup := range policyApplicationBackup {
+			// Get the updated status if we see it as not completed
+			if !s.isApplicationBackupComplete(backup.Status) {
+				pendingApplicationBackupStatus, err := getApplicationBackupStatus(backup.Name, backupSchedule.Namespace)
+				if err != nil {
+					s.Recorder.Event(backupSchedule,
+						v1.EventTypeWarning,
+						string(stork_api.ApplicationBackupStatusFailed),
+						fmt.Sprintf("Error getting status of backup %v: %v", backup.Name, err))
+				}
```

Comment:
Should we return err if we could not get the backup status? 
Currently we are marking the backup as failed if the status cannot be retrieved.

Reply:
Added a continue for errors other than `NotFound` because we want to mark those as failed and still check status for other backups